src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,41 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    public static final int STORAGE_SIZE = 10;
+    private int size = 0;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[STORAGE_SIZE];
+        values = (V[]) new Object[STORAGE_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] != null && keys[i].equals(key) || keys[i] == null && key == null) {
```

Comment:
```suggestion
            if (keys[i] != null && keys[i].equals(key) || keys[i] == key) {
```
extract the logic of iteratin and comparing - it's common for both methods, put and get. Or at least extract the comparison logic